test_util/aws.py

```diff
@@ -258,7 +258,16 @@ def create(cls, stack_name, boto_wrapper, template_url,
             'PrivateAgentInstanceType': private_agent_type,
             'PrivateSubnet': private_subnet}
         stack = boto_wrapper.create_stack(stack_name, template_url, parameters)
-        return cls(stack.stack.stack_name, boto_wrapper), SSH_INFO['coreos']
+        try:
+            os_string = template_url.split('/')[-1].split('.')[-2].split('-')[0]
+            ssh_info = CF_OS_SSH_INFO[os_string]
+        except KeyError:
+            logging.exception('No SSH info found corresponding to OS string: {}'.format(os_string))
```

Comment:
It'd be handy to have the template URL here as well since this error can also be caused by an unexpected URL format. You can use the same handler for both exceptions since they're describing the same error (unexpected template URL).